client/boptest.py

```diff
@@ -0,0 +1,262 @@
+        Purpose: retrieving the haystack tagged data, 
+              which are model-exchange variables in FMU. 
+        Inputs:  a json file containing all the tagged data.
+        Returns: a list for all the tagged data, with the tagging properties
+              like id, dis, site_ref, etc...
+        '''
+        with open( tag_filepath ) as json_data:
+            tag_data = json.load(json_data)
+     
+        return tag_data
+
+
+    #This is the function to create tags
+    def create_DisToID_dictionary(self, tag_filepath):
+        '''
+        Purpose: matching the haystack display-name and IDs
+        Inputs:  a json file containing all the tagged data
+        Returns: a dictionary mathching all display-names and IDs
+             a dictionary matching all inputs and IDs
+             a dictionary matching all outputs and IDs 
+             a dictionary matching all IDs and display-names 
+        '''
+        dis_and_ID={}
+        inputs_and_ID = {}
+        outputs_and_ID = {}
+        id_and_dis={}
+
+        tag_data = get_tag_data(tag_filepath)
+
+        for point in tag_data:
+            var_name = point['dis'].replace('s:','')
+            var_id = point['id'].replace('r:','')
+
+            dis_and_ID[var_name] = var_id
+            id_and_dis[var_id] = var_name
+
+            if 'writable' in point.keys():
+                #it means it is a input variable.
+                #clean the var-name, discarding: ':input','s:','r:'
+                #input_var = var_name.replace(':input','')
+                #input_var = input_var.replace('s:','')
+                inputs_and_ID[var_name] = var_id
+
+            if 'writable' not in point.keys() and 'point' in point.keys():
+                #it means it is an output variable, plus not sitetag.
+                #clean the var-name, discarding: ':output','s:','r:'
+                #output_var = var_name.replace(':output','')
+                #output_var = output_var.replace('s:','')
+                outputs_and_ID[var_name] = var_id
+
+        return (dis_and_ID, inputs_and_ID, outputs_and_ID, id_and_dis)
+
+
+
+    # Stop a simulation for model identified by id
+    def stop(self, id):
+        pass
+
+    # Return the input values for simulation identified by id,
+    # in the form of a dictionary. See setInputs method for dictionary format
+    def inputs(self, id):
+        pass
+    # Set inputs for model identified by id
+    # The inputs argument should be a dictionary of 
+    # with the form
+    # inputs = {
+    #  input_name: { level_1: value_1, level_2: value_2...},
+    #  input_name2: { level_1: value_1, level_2: value_2...}
+    # }
+    def setInputs(self, id, **inputs):
+        pass
+    # Return a dictionary of the output values
+    # result = {
+    # output_name1 : output_value1,
+    # output_name2 : output_value2
+    #}
+    def outputs(self, id):
+        pass
+
+
+###############################################################################
```

Comment:
The following lines are better in their own test file, leaving the Baptest class file clean. Otherwise every time we try to import the class, it gets instantiated.